streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableValueGetter.java

```diff
@@ -25,5 +25,21 @@
 
     ValueAndTimestamp<V> get(K key);
 
+    /**
+     * Returns the latest record version, associated with the provided key, with timestamp
+     * not exceeding the provided timestamp bound. This method may only be called if
+     * {@link #isVersioned()} is true.
+     */
+    default ValueAndTimestamp<V> get(K key, long asOfTimestamp) {
+        throw new UnsupportedOperationException("get(key, timestamp) is only supported for versioned stores");
+    }
+
+    /**
+     * @return whether this value getter supports multiple record versions for the same key.
+     *         If true, then {@link #get(Object, long)} must be implemented. If not, then
+     *         {@link #get(Object, long)} must not be called.
+     */
+    boolean isVersioned();
```

Comment:
As `get(key, asOfTs)` throws and exception by default, should we let `isVersioned` return `false` by default?

Reply:
Sure, I was on the fence about this. The benefit of not having a default value is that it forces all new implementations to think about whether versioning can/should be supported, but I'm not too worried since it's an internal interface. If you prefer having the default, I will make the update.

Reply:
The same argument seems to apply to `get()` and it does have a default? Seems inconsistent to me? But I agree, it's internal so also don't feel strong about it. Leave it to your judgement.

Reply:
Heh, I made one of non-default in order to force future authors of new value getters to have to consider at least one of them, with the hope that considering one implies considering the other as well :) I will leave it as-is for now, can always be changed in the future.